src/controllers/authControllers.js

```diff
@@ -64,51 +94,178 @@ exports.signup = async (req, res) => {
+    });
+
+    const to = user.email;
+    const subject = "Activate your Oral Moral's Account Now";
+    const html = `<p>You're just one click away from getting started with Oral Morals. All you need to do is verify your email address to activate your Oral Morals account. Click <a href="http://localhost:${process.env.PORT}/verify/${newToken.token}">here</a></p>`;
+
+    // sendEmail(to, subject, html);
+
+    res.status(200).json({ message: `A verification email has been sent to ${user.email}.` });
+  } catch (err) {
+    console.log(err);
+    if (err) return res.status(500).json({ message: "Something went wrong. Please try again later" });
+  }
+};
+
 exports.login = async (req, res) => {
   try {
-    const user = await User.findOne({ email: req.body.email });
+    const { email, password } = req.body;
+
+    // Search for existing user by email
+    const user = await User.findOne({ email });
     if (!user)
-      return res.status(404).json({
-        message: "User does not exist",
+      return res.status(401).json({
+        message: `The email ${user.email} is not associated with any account. Double-check your email and try again.`,
       });
 
-    // If user
+    // If existing user is found, compare passwords
+    const match = await bcrypt.compare(password, user.password);
+    if (!match) return res.status(400).json({ message: "Incorrect Email or Password" });
+
+    const token = jwt.sign(
+      {
+        id: user._id,
+        email: user.email,
+        dateOfBirth: user.dateOfBirth,
+        username: user.username,
+      },
+      process.env.JWT_SECRET,
+      { expiresIn: "24h" }
+    );
+    // If passwords match, log in the user
+    return res.status(200).json({ message: `${user.email} logged in successfully!`, token });
   } catch (error) {
     console.log(error);
     res.status(500).json({ success: false, message: error.message });
   }
 };
 
-exports.authorizeUser = async (req, res, next) => {
+exports.passwordResetRequest = async (req, res) => {
   try {
-    // Check if authorization header credentials were sent with request.
-    if (!req.headers.authorization) return res.status(401).json({ message: "Authorization header required" });
+    let user = await User.findOne({ email: req.body.email });
 
-    // Check if the authorization type is Bearer.
-    if (!req.headers.authorization.startsWith("Bearer "))
-      return res.status(401).json({ message: "Authorization format is: Bearer <token>" });
+    let token = await TokenModel.create({
+      userID: user._id,
+      token: uuidv4(),
+      expiresIn: moment().add(1, "hours"),
+      tokenType: "password-reset",
+    });
 
-    // Get the token from the authorization header by removing "Bearer " from the string.
-    let token = req.headers.authorization.replace("Bearer ", "");
+    const to = user.email;
+    const subject = "Fogot your password?";
+    const html = `<p>Click <a href="reset-form-with-token>here</a> to reset your password.</p>`;
```

Comment:
The href does not link to anywhere but let see what design does before we make any changes to this flow.